### PR TITLE
Uncompress message payload by content_encoding meta data

### PR DESCRIPTION
**Pull request by bautrey37**
There was lots of trouble getting the service to run the tests. I would get an error about hackney unable to start up. The usual fix was to clean all dependencies and reinstall everything, and maybe one run of `mix test` would work. I created a CI pipeline that can run the tests more reliably than local dev environment. 

My `.tool-versions` file:
```
erlang 21.0
elixir 1.7.1-otp-21
```

Run these commands to make sure it is installed corrected:
- `mix deps.clean --all`
- `rm -rf _build`
- `mix deps.get`
- `mix deps.compile -f`
- `mix compile -f`
- `mix test`

Somtimes upgrading `amqp` to `1.0.3` will help fix the issue locally, but not always. 

EN-5832

### CODE REVIEW

File: test/task_bunny/message_test.exs [hunk 1]
```diff
@@ -36,6 +36,14 @@ defmodule TaskBunny.MessageTest do
 
       assert {:error, :job_not_loaded} == Message.decode(message)
     end
+
+    test "decode compressed json" do
+      encoded_json = %{"pay" => 'load'} |> Poison.encode!()
+      compressed_json = :zlib.compress(encoded_json)
+
+      message = Message.uncompress(compressed_json, %{content_encoding: "zlib", some: "thing"})
```
Comment (urmastalimaa) [inline test/task_bunny/message_test.exs:44]: The `some: "thing"` part can IMO be dropped

File: mix.exs [hunk 2]
```diff
@@ -70,6 +72,12 @@ defmodule TaskBunny.Mixfile do
     ]
   end
 
+  defp aliases do
+    [
+      "test.ci": ["test --color --max-cases=10"]
```
Comment (urmastalimaa) [inline mix.exs:77]: Why was max-cases necessary to be specified?

File: test/task_bunny/message_test.exs (hunk 1 above)
Comment (bautrey37) [reply test/task_bunny/message_test.exs:44]: True, I just wanted to say that other parameters can also exist. I removed it.

File: mix.exs (hunk 2 above)
Comment (bautrey37) [reply mix.exs:77]: Its not necessary to set. It was from an online guide.

**Review by urmastalimaa — APPROVED** [review]

**Review by riiwo — APPROVED** [review]
I am not going to block you for this but a couple of remarks about the pull request.

Some of the commit messages are lacking the _why_ and _what_. We have guide how to write commit messages and it can be found here: https://salemove.atlassian.net/wiki/spaces/ENG/pages/15728856/Style+Guide+Conventions#StyleGuide%26Conventions-GitCommitMessages
 The general idea being documenting the decisions made during the development.
 
 The other nitpick would be this commit: https://github.com/salemove/task_bunny/pull/2/commits/30cd6cdb05b5d1aea0dcfdd553d3e7fb24382ad8 as these changes do not really deserve themselves a separate commit (it is like fixing the feature in another commit but it is still the same feature) and maybe should be together with the previous one.

I constantly fail with it myself, but let us improve together 😃

**Pull request merged**